specifyweb/frontend/js_src/lib/components/InitialContext/treeRanks.ts

```diff
@@ -116,22 +106,61 @@ function getTreeScope(
   );
 }
 
+export function getTreeDefinitions<TREE_NAME extends AnyTree['tableName']>(
+  tableName: TREE_NAME,
+  treeDefinitionId: number | undefined | 'all' = getCache(
+    'tree',
+    `definition${tableName}`
+  )
+): typeof treeDefinitions[TREE_NAME] {
```

Comment:
(optional) Would `TreeInformation[TREE_NAME]` work here instead? If so, there's a couple more places I noticed with `typeof treeDefinitions` that can be replaced.